index.bs

```diff
@@ -135,7 +135,25 @@ Issue: Monkeypatch `window.open()`.
 
 # Fetch monkeypatches # {#fetch-monkeypatches}
 
-Issue: Specify monkeypatches for source/trigger registration.
+<pre class="idl">
+dictionary AttributionReportingRequestOptions {
+  required boolean eventSourceEligible;
+  required boolean triggerEligible;
+};
+
+partial dictionary RequestInit {
+  AttributionReportingRequestOptions attributionReporting;
+};
+
+partial interface XMLHttpRequest {
+  [RaisesException, SecureContext]
+  undefined setAttributionReporting(AttributionReportingRequestOptions options);
+};
+</pre>
+
+If {{AttributionReportingRequestOptions}} is not set, then no
```

Comment:
This part should be a monkey patch on the Fetching processing model.
e.g. "In [http network or cache fetch](https://fetch.spec.whatwg.org/#http-network-or-cache-fetch) after step 8.14, call "Append attribution reporting", where "append attribution reporting" is an algorithm defined here which adds the header when conditions are met, and flips on a boolean that enabled trigger registration with the response.

Something similar should be done for XHR

Reply:
Yes, we do need to integrate with `fetch` itself, but that's true even without this change.

Reply:
Is that something you're interested in doing as a followup? IME, it's better to start out early (and small) and expand from there.

Reply:
I plan to do it, but we also need to monkeypatch navigation (also even without this change), and I don't think doing so should block getting this landed as soon as possible, though I will take a look at least the fetch part now, while we're still waiting for feedback on this.